libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -997,16 +997,49 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		break;
 	case FunctionType::Kind::Internal:
 	{
-		FunctionDefinition const* functionDef = ASTNode::resolveFunctionCall(_functionCall, &m_context.mostDerivedContract());
-
+		solAssert(functionType);
 		solAssert(!functionType->takesArbitraryParameters());
 
 		vector<string> args;
-		if (functionType->hasBoundFirstArgument())
-			args += IRVariable(_functionCall.expression()).part("self").stackSlots();
+		FunctionDefinition const* functionDef = nullptr;
 
-		for (size_t i = 0; i < arguments.size(); ++i)
-			args += convert(*arguments[i], *parameterTypes[i]).stackSlots();
+		if (!_functionCall.isSuffixCall() || parameterTypes.size() == 1)
+		{
+			functionDef = ASTNode::resolveFunctionCall(_functionCall, &m_context.mostDerivedContract());
+
+			if (functionType->hasBoundFirstArgument())
+				args += IRVariable(_functionCall.expression()).part("self").stackSlots();
+
+			for (size_t i = 0; i < arguments.size(); ++i)
+				args += convert(*arguments[i], *parameterTypes[i]).stackSlots();
+		}
+		else
+		{
+			functionDef = dynamic_cast<FunctionDefinition const*>(&functionType->declaration());
+			solAssert(functionDef);
+			solAssert(!functionDef->virtualSemantics());
+			solAssert(!functionType->hasBoundFirstArgument());
+
+			solAssert(parameterTypes.size() == 2);
+			solAssert(arguments.size() == 1 && arguments[0]);
+			auto&& literal = dynamic_cast<Literal const*>(arguments[0].get());
+			solAssert(literal);
+			solAssert(literal->annotation().type);
+
+			auto&& literalRationalType = dynamic_cast<RationalNumberType const*>(literal->annotation().type);
+			solAssert(literalRationalType);
+
+			auto&& [mantissa, exponent] = literalRationalType->fractionalDecomposition();
+			solAssert(mantissa && exponent);
+
+			IRVariable mantissaVar(m_context.newYulVariable(), *mantissa);
+			define(mantissaVar) << toCompactHexWithPrefix(mantissa->literalValue(literal)) << "\n";
+			args += convert(mantissaVar, *parameterTypes[0]).stackSlots();
+
+			IRVariable exponentVar(m_context.newYulVariable(), *exponent);
+			define(exponentVar) << toCompactHexWithPrefix(exponent->literalValue(literal)) << "\n";
+			args += convert(exponentVar, *parameterTypes[1]).stackSlots();
```

Comment:
Good suggestion. Looks like it can. We validate that integers are in the range matching the type so we should be fine without any runtime conversion there.